src/app/dataset-block/metadata-block/components/event-details/components/common/yaml-event-viewer/yaml-event-viewer.component.ts

```diff
@@ -29,21 +32,37 @@ export class YamlEventViewerComponent {
         Object.entries(this.event).forEach(([key, value]) => {
             key !== "__typename" && value ? (result += `${key}:\n`) : "";
             if (value && key !== "__typename") {
-                Object.entries(value as object).forEach(([property, data]) => {
-                    if (data) {
-                        result += `\u00A0\u00A0${
-                            property === "__typename" ? "kind" : property
-                        }: ${
-                            property === "__typename"
-                                ? this.kindMapper[data as string]
-                                : property === "queries"
-                                ? (data as SqlQueryStep[]).map(
-                                      (item) => `>\n${item.query}`,
-                                  )
-                                : (data as string)
+                if (key === "inputs") {
+                    (value as TransformInput[]).map((item: TransformInput) => {
+                        result += `\u00A0\u00A0- kind: ${
+                            this.kindMapper[item.dataset.__typename as string]
                         }\n`;
-                    }
-                });
+                        result += `\u00A0\u00A0\u00A0\u00A0id: ${item.dataset.id}\n`;
```

Comment:
Very ugly. Unreadable. Declare constants for 1 level tabs and 2 level tabs. I don't think we will ever have 3 or more.

Reply:
Fixed